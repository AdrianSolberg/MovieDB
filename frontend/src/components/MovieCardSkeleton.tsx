import { Card, CardContent, CardHeader } from "@/shadcn/components/ui/card";
import { Skeleton } from "@/shadcn/components/ui/skeleton";

const MovieCardSkeleton = () => {
    return (
        <Card className="m-0 shadow-lg hover:cursor-pointer hover:shadow-slate-600">
            <CardHeader className="aspect-[2/3] p-2">
                <Skeleton className="h-full" />
            </CardHeader>
            <CardContent className="flex flex-col items-center px-2 pb-2">
                <Skeleton className="h-5 w-1/2" />
                <Skeleton className="h-5 w-5/6" />
            </CardContent>
        </Card>
    );
};

export default MovieCardSkeleton;
