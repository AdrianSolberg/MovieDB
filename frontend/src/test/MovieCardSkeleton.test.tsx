import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import MovieCardSkeleton from "@/components/MovieCardSkeleton";

describe("MovieCardSkeleton", () => {
    it("matches snapshot", () => {
        const { asFragment } = render(<MovieCardSkeleton />);
        expect(asFragment()).toMatchSnapshot();
    });
});
