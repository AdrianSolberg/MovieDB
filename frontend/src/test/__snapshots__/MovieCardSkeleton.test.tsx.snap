// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCardSkeleton > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground m-0 shadow-lg hover:cursor-pointer hover:shadow-slate-600"
  >
    <div
      class="flex flex-col space-y-1.5 aspect-[2/3] p-2"
    >
      <div
        class="animate-pulse rounded-md bg-muted h-full"
      />
    </div>
    <div
      class="p-6 pt-0 flex flex-col items-center px-2 pb-2"
    >
      <div
        class="animate-pulse rounded-md bg-muted h-5 w-1/2"
      />
      <div
        class="animate-pulse rounded-md bg-muted h-5 w-5/6"
      />
    </div>
  </div>
</DocumentFragment>
`;
